# 01. Data Understanding - Berlin Housing Market Analysis
**Objectives**
Explore the structure and basic characteristis of the Berlin housing

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

pd.set_option('display.max_columns', None)
sns.set_style('whitegrid')
plt.rcParams['figure.figsize']= (10, 6)

rentals= pd.read_csv('../data/raw/rentals.csv')

print("Rentals dataset loaded successfully")
print("Shape:", rentals.shape)
print("\nColumns:")
print(rentals.columns.tolist())

Rentals dataset loaded successfully
Shape: (30000, 30)

Columns:
['id', 'date_listed', 'ortsteil', 'bezirk', 'kiez_premium', 'lat', 'lon', 'property_type', 'rooms', 'area_m2', 'floor', 'total_floors', 'building_era', 'energy_class', 'position', 'condition', 'has_lift', 'has_balcony', 'furnished', 'transit_station', 'transit_line', 'transit_distance_min', 'to_brandenburg_gate_km', 'kaltmiete_eur_monthly', 'nebenkosten_eur_monthly', 'warmmiete_eur_monthly', 'rent_per_m2_kalt_eur', 'rent_includes_warmmiete', 'kaution_months', 'warmmiete_eur_monthly_usd']


In [4]:
print("===Basic Info ===")
print(rentals.info())


===Basic Info ===
<class 'pandas.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 30 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   id                         30000 non-null  str    
 1   date_listed                30000 non-null  str    
 2   ortsteil                   30000 non-null  str    
 3   bezirk                     30000 non-null  str    
 4   kiez_premium               30000 non-null  str    
 5   lat                        30000 non-null  float64
 6   lon                        30000 non-null  float64
 7   property_type              30000 non-null  str    
 8   rooms                      30000 non-null  int64  
 9   area_m2                    30000 non-null  float64
 10  floor                      30000 non-null  int64  
 11  total_floors               30000 non-null  int64  
 12  building_era               30000 non-null  str    
 13  energy_class               30000 non-nu

In [5]:
print("=== First 5 rows ===")
rentals.head() 

=== First 5 rows ===


,id,date_listed,ortsteil,bezirk,kiez_premium,lat,lon,property_type,rooms,area_m2,floor,total_floors,building_era,energy_class,position,condition,has_lift,has_balcony,furnished,transit_station,transit_line,transit_distance_min,to_brandenburg_gate_km,kaltmiete_eur_monthly,nebenkosten_eur_monthly,warmmiete_eur_monthly,rent_per_m2_kalt_eur,rent_includes_warmmiete,kaution_months,warmmiete_eur_monthly_usd
0,R000001,2023-12-18,Adlershof,Treptow-Köpenick,medium,52.44391,13.53835,2_room,2,55.1,0,3,post_war_1949_1990,E,vorderhaus,kernsaniert,False,True,False,Konstanzer Straße,U7,203,13.53,517,281,798,9.37,False,3,861
1,R000002,2021-06-07,Lankwitz,Steglitz-Zehlendorf,medium,52.44035,13.33848,4_room,4,106.6,5,6,modern_1990_2010,E,vorderhaus,saniert,True,False,False,Heinrich-Heine-Straße,U8,111,8.85,1281,347,1628,12.03,True,3,1758
2,R000003,2022-08-19,Britz,Neukölln,low,52.44825,13.45215,3_room,3,102.5,3,5,modern_1990_2010,C,vorderhaus,saniert,True,False,False,Neukölln,U7,27,9.09,1366,524,1889,13.32,False,1,2041
3,R000004,2025-12-29,Hellersdorf,Marzahn-Hellersdorf,low,52.53063,13.62405,3_room,3,90.5,0,5,altbau_pre_1949,G,seitenflügel,kernsaniert,True,False,False,Neukölln S,S Ringbahn,167,16.74,469,335,804,5.18,True,2,869
4,R000005,2022-07-18,Wartenberg,Lichtenberg,low,52.58212,13.50497,2_room,2,55.6,5,6,modern_1990_2010,H,vorderhaus,modernisiert,True,True,False,Gesundbrunnen S,S Ringbahn,110,11.30,369,260,629,6.64,False,2,679


In [7]:
print("=== Missing Values ===")
missing = rentals.isnull().sum()
print(missing[missing >0])
print("\nTotal missing values:")

=== Missing Values ===
Series([], dtype: int64)

Total missing values:


In [8]:
print("=== Numerical Summary ===")
rentals.describe().round(2) 

=== Numerical Summary ===


,lat,lon,rooms,area_m2,floor,total_floors,transit_distance_min,to_brandenburg_gate_km,kaltmiete_eur_monthly,nebenkosten_eur_monthly,warmmiete_eur_monthly,rent_per_m2_kalt_eur,kaution_months,warmmiete_eur_monthly_usd
count,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00,30000.00
mean,52.51,13.39,2.52,75.59,2.48,4.95,89.23,10.11,946.80,265.89,1212.69,12.12,2.50,1309.70
std,0.06,0.13,1.13,37.47,1.86,1.38,72.85,4.61,724.03,142.68,829.22,5.48,0.67,895.55
min,52.36,13.11,1.00,20.00,0.00,3.00,0.00,0.23,86.00,50.00,148.00,2.55,1.00,160.00
25%,52.46,13.30,2.00,48.40,1.00,4.00,27.00,6.65,458.00,164.00,643.00,8.18,2.00,694.00
50%,52.51,13.39,2.00,68.70,2.00,5.00,75.00,9.69,735.00,237.00,988.00,10.78,3.00,1067.00
75%,52.56,13.48,3.00,94.10,4.00,6.00,137.00,13.21,1203.00,332.00,1532.00,14.80,3.00,1655.00
max,52.66,13.71,5.00,302.30,8.00,8.00,424.00,26.09,7624.00,1305.00,8685.00,46.40,3.00,9380.00


In [9]:
print("=== Unique values in key categorical columns ===")
print("Districts (Bezirk):", rentals['bezirk'].nunique())
print(rentals['bezirk'].value_counts().head(10))

print("\nProperty Type:")
print(rentals['property_type'].value_counts())

print("\nKiez Premium:")
print(rentals['kiez_premium'].value_counts())

=== Unique values in key categorical columns ===
Districts (Bezirk): 12
bezirk
Reinickendorf                 4124
Treptow-Köpenick              3510
Pankow                        2929
Lichtenberg                   2894
Spandau                       2639
Charlottenburg-Wilmersdorf    2573
Steglitz-Zehlendorf           2514
Mitte                         2190
Tempelhof-Schöneberg          2173
Neukölln                      1840
Name: count, dtype: int64

Property Type:
property_type
2_room         10218
3_room          7529
1_room          5992
4_room          4640
5_room_plus     1621
Name: count, dtype: int64

Kiez Premium:
kiez_premium
medium    12332
low       11111
high       6557
Name: count, dtype: int64


In [10]:
print("=== Rent per m2 (Kalt) ===")
print(rentals['rent_per_m2_kalt_eur'].describe().round(2))

=== Rent per m2 (Kalt) ===
count    30000.00
mean        12.12
std          5.48
min          2.55
25%          8.18
50%         10.78
75%         14.80
max         46.40
Name: rent_per_m2_kalt_eur, dtype: float64


### Initial Observations
- The dataset is clean (no missing values).
- Average kaltmiete is around 947 euros with an average size of 76m2.
- Average rent per m2 is app 12.12 euros.
- Reinikendorf, Treptow and Pankow have the highest number of listings.
- There is a wide range in both sieze and rent due to luxury apartments.